test/over_react/component/shared_error_boundary_tests.dart

```diff
@@ -102,26 +114,605 @@ void sharedErrorBoundaryTests(BuilderOnlyUiFactory builder) {
+            final secondError = calls[0]['onComponentDidCatch'][0];
+
+            expect('$firstError', isNot('$secondError'), reason: 'test setup sanity check');
+            _setCallbackVarValues();
+          });
+
+          test('the components wrapped by the ErrorBoundary get remounted', () {
+            expect(getByTestId(jacket.getInstance(), 'nestedFlawedComponent'), isNotNull,
+                reason: 'The flawed component should have been remounted');
+            expect(jacket.getDartInstance().state.showFallbackUIOnError, isFalse,
+                reason: 'Fallback UI should be not be rendered');
+          });
+
+          test('the correct callbacks are called with the correct arguments', () {
+            expect(calls.any((call) => call.keys.single == 'onComponentIsUnrecoverable'), isFalse,
+                reason: 'onComponentIsUnrecoverable should not have been called');
+
+            expect(errorSentToComponentDidCatchCallback, isA<FlawedComponentException2>());
+            expect(errorInfoSentToComponentDidCatchCallback, isA<ReactErrorInfo>());
+          });
+        });
+
+        group('and they occurred less frequently than the value of props.identicalErrorFrequencyTolerance:', () {
+          setUp(() async {
+            getFlawedButtonNode().click();
+            expect(calls.length, 1, reason: 'test setup sanity check');
+            expect(calls.single.keys.single, isNot('onComponentIsUnrecoverable'), reason: 'test setup sanity check');
+            final firstError = calls[0]['onComponentDidCatch'][0];
+
+            calls.clear();
+            await new Future.delayed(const Duration(milliseconds: identicalErrorFrequencyToleranceInMs + 50));
+
+            getNestedFlawedButtonThatThrowsADifferentErrorNode().click();
+            final secondError = calls[0]['onComponentDidCatch'][0];
+
+            expect('$firstError', isNot('$secondError'), reason: 'test setup sanity check');
+            _setCallbackVarValues();
+          });
+
+          test('the components wrapped by the ErrorBoundary get remounted', () {
+            expect(getByTestId(jacket.getInstance(), 'nestedFlawedComponent'), isNotNull,
+                reason: 'The flawed component should have been remounted');
+            expect(jacket.getDartInstance().state.showFallbackUIOnError, isFalse,
+                reason: 'Fallback UI should be not be rendered');
+          });
+
+          test('the correct callbacks are called with the correct arguments', () {
+            expect(calls.any((call) => call.keys.single == 'onComponentIsUnrecoverable'), isFalse,
+                reason: 'onComponentIsUnrecoverable should not have been called');
+
+            expect(errorSentToComponentDidCatchCallback, isA<FlawedComponentException2>());
+            expect(errorInfoSentToComponentDidCatchCallback, isA<ReactErrorInfo>());
+          });
+        });
+      });
     });
   });
+
+//  group('throws a PropError when', () {
```

Comment:
Should we add a TODO here to mention that the validate props needs to be added back?